src/deadline/client/cli/_groups/manifest_group.py

```diff
@@ -0,0 +1,726 @@
+    metadata["Metadata"]["file-system-location-name"] = manifest_file
+
+    bucket_name: str = ""
+    manifest_path: str = ""
+    session: boto3.Session = api.get_boto3_session()
+    if not cas_path:
+        farm_id = config_file.get_setting("defaults.farm_id", config=config)
+        queue_id = config_file.get_setting("defaults.queue_id", config=config)
+
+        deadline = api.get_boto3_client("deadline", config=config)
+        queue = deadline.get_queue(
+            farmId=farm_id,
+            queueId=queue_id,
+        )
+        bucket_name = queue["jobAttachmentSettings"]["s3BucketName"]
+        manifest_path = queue["jobAttachmentSettings"]["rootPrefix"]
+
+        # IF we supplied a farm and queue, use the queue credentials.
+        session = api.get_queue_user_boto3_session(
+            deadline=deadline,
+            config=config,
+            farm_id=farm_id,
+            queue_id=queue_id,
+            queue_display_name=queue["displayName"],
+        )
+
+    else:
+        # Self supplied cas path.
+        url_fragments = urlparse(cas_path)
+        bucket_name = url_fragments.netloc
+        manifest_path = url_fragments.path
+
+    logger.echo(f"Uploading Manifest to {bucket_name} {manifest_path}")
+
+    # Always upload the manifest file to case root /Manifest with the original file name.
+    manifest_path = manifest_path + "/Manifests/" + Path(manifest_file).name
+
+    # Uploader
+    upload = S3AssetUploader(session=session)
+    with open(manifest_file) as manifest:
+        upload.upload_bytes_to_s3(
+            bytes=BytesIO(manifest.read().encode("utf-8")),
+            bucket=bucket_name,
+            key=manifest_path,
+            progress_handler=logger.echo,
+            extra_args=metadata,
+        )
+
+    logger.echo("Uploading successful!")
+
+
+def read_local_manifest(manifest: str) -> BaseAssetManifest:
+    """
+    Read manifests specified by filepath to manifest folder, returns BaseAssetManifest Object
+    """
+    input_files = glob.glob(os.path.join(manifest, "*_input"))
+
+    if not input_files:
+        raise ValueError(f"No manifest files found in {manifest}")
+    elif len(input_files) >= 2:
+        raise NonValidInputError(
+            f"Multiple input manifest files are not supported, found: {input_files}."
+        )
+
+    manifest_file_path = input_files[0]
+
+    with open(manifest_file_path) as input_file:
+        manifest_data_str = input_file.read()
+        asset_manifest = decode_manifest(manifest_data_str)
+
+        return asset_manifest
+
+
+def clear_S3_mapping(manifest: str):
+    """
+    Clears manifest_s3_mapping file contents if it previously exists.
+    """
+    for filename in os.listdir(manifest):
+        if filename.endswith("manifest_s3_mapping"):
+            # if S3 mapping already exists, clear contents
+            filepath = os.path.join(manifest, filename)
+            with open(filepath, "w") as _:
+                pass
+
+
+def diff_manifest(
```

Comment:
For the manifest helpers, do they make sense to be in the CLI group file? What about grouping them to a file under attachment folder for potential code reuse?